provider/pkg/provider/provider.go

```diff
@@ -2756,10 +2756,11 @@ func (k *kubeProvider) tryServerSidePatch(
 		return nil, nil, false, err
 	}
 	if se, isStatusError := err.(*errors.StatusError); isStatusError {
-		// If the resource field is immutable.
 		if se.Status().Code == http.StatusUnprocessableEntity ||
```

Comment:
Are you sure it can't return 422 for another reason? I would think e.g., admission controllers could respond with a 422.

Reply:
Maybe this is meant to be `&&` here -- it's 422 _and_ the message mentions immutable fields.

Reply:
Hmm, I think you're right about the compound condition. It's been like that for over a year, but I guess the usage may not have been high enough to uncover edge cases with it yet. I'll change it to `&&`.

Reply:
I noticed there are some predicates over errors in "k8s.io/apimachinery/pkg/api/errors", may be useful for narrowing, or clarifying, the condition here.

Reply:
> I noticed there are some predicates over errors in "k8s.io/apimachinery/pkg/api/errors", may be useful for narrowing, or clarifying, the condition here.

It doesn't look like they have any that are specific to `StatusUnprocessableEntity` or immutable fields, unfortunately.